cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1336,41 +1376,309 @@ collectTxBodySimpleScripts TxBodyContent {
 
     simpleScriptInEra _ = []
 
+
 toShelleySimpleScript :: SimpleScriptInEra era
                       -> Ledger.Script (ShelleyLedgerEra era)
 toShelleySimpleScript (SimpleScriptInEra langInEra version script) =
     toShelleyScript (ScriptInEra langInEra (SimpleScript version script))
 
-toShelleyWithdrawal :: [(StakeAddress, Lovelace, a)] -> Shelley.Wdrl StandardCrypto
+
+getShelleyTxBodyContent ::  Shelley.TxBody (ShelleyLedgerEra ShelleyEra)
+                        ->  Maybe
```

Comment:
Formatting. Also why do we need ` Maybe (Shelley.Metadata (ShelleyLedgerEra ShelleyEra))`? This function should also be parameterized by `ShelleyBasedEra era` so  that we don't need to have a separate one for each era.

Reply:
1. This parameter is used to fill the `txMetadata` field.
2. These functions seem pretty different. Okay, I'll try to generalize.

Reply:
@cblp I think I understand why you are supplying the tx metadata in the function arguments. We shouldn't do this. The txbody stores the auxiliary data hash, we should display that instead.

Reply:
1. Removed txMetadata logic. I don't understand why it is a problem.
2. `metadata` parameter is still used to check hash.
3. Should I add metadata hash display now?